scripts/docs/extension_instructions.md

```diff
@@ -10,154 +10,91 @@ working with the same Bareflank workspace directory structure.
 
 ## Building and Integrating Extensions
 Extensions can be added to the Bareflank build system by using the
-*vmm_extension()* macro within a Bareflank build configuration file
-(bfconfig.cmake). The vmm_extension macro is compatible with all valid arguments
-to CMake's built-in
-[ExternalProject_Add()](https://cmake.org/cmake/help/v3.6/module/ExternalProject.html)
-function. For example, you could add the Bareflank extended apis extension from
-GitHub by adding the following to your build configuration file
-(bfconfig.cmake):
+`EXTENSION` variable to pass on the root directories of the extensions.
 
+In the top-level *CMakeLists.txt* file, you can setup and include extensions by
+using the `vmm_extension()` function. The vmm_extension function is a wrapper around
+the `add_subproject()` function, which makes it compatible with all valid arguments
+to CMake's built-in [ExternalProject_Add()](https://cmake.org/cmake/help/v3.6/module/ExternalProject.html)
+function.
+Here is an example for adding the extended_apis extension.
+
+First, you need to clone the extension from GitHub:
+
+```bash
+cd ~/bareflank
+git clone https://github.com/bareflank/extended_apis.git
 ```
-vmm_extension(
-    extended_apis
-    GIT_REPOSITORY https://github.com/bareflank/extended_apis.git
-    GIT_TAG dev
+
+After cloning the extension, you can include it in the build by passing
+it's root directory to CMake and defining the default VMM when configuring.
+There are two ways to accomplish this.
+Either pass it diretcly via the command-line `-DDEFAULT_VMM=eapis_vmm -DEXTENSION=../extended_apis`, or do it
+in the build configuration file `~/bareflank/build/config.cmake`:
```

Comment:
I would get rid of the example config stuff as this is more advanced. Instead, I would point the user to the example config if they want to use a config.